src/test_l2_tvSettings.c

```diff
@@ -64,50 +63,3640 @@
+    status = EnableWBCalibrationMode(true);
+    UT_LOG_DEBUG("Return status: %d", status);
+    UT_ASSERT_EQUAL(status, tvERROR_NONE);
+    if (status != tvERROR_NONE)
+    {
+        UT_LOG_ERROR("EnableWBCalibrationMode failed with status: %d", status);
+    }
+
+    UT_LOG_DEBUG("Invoking GetCurrentWBCalibrationMode()");
+    status = GetCurrentWBCalibrationMode(&value);
+    UT_LOG_DEBUG("Return status: %d, Output value: %d", status, value);
+    UT_ASSERT_EQUAL(status, tvERROR_NONE);
+    UT_ASSERT_EQUAL(value, true);
+    if (status != tvERROR_NONE || value != true)
+    {
+        UT_LOG_ERROR("GetCurrentWBCalibrationMode failed with status=%d", status);
+    }
+
+    UT_LOG_DEBUG("Invoking EnableWBCalibrationMode() with value set to false");
+    status = EnableWBCalibrationMode(false);
+    UT_LOG_DEBUG("Return status: %d", status);
+    UT_ASSERT_EQUAL(status, tvERROR_NONE);
+    if (status != tvERROR_NONE)
+    {
+        UT_LOG_ERROR("EnableWBCalibrationMode failed with status: %d", status);
+    }
+
+    UT_LOG_DEBUG("Invoking GetCurrentWBCalibrationMode()");
+    status = GetCurrentWBCalibrationMode(&value);
+    UT_LOG_DEBUG("Return status: %d, Output value: %d", status, value);
+    UT_ASSERT_EQUAL(status, tvERROR_NONE);
+    UT_ASSERT_EQUAL(value, false);
+    if (status != tvERROR_NONE || value != false)
+    {
+        UT_LOG_ERROR("GetCurrentWBCalibrationMode failed with status=%d", status);
+    }
+
+    UT_LOG_DEBUG("Invoking TvTerm()");
+    status = TvTerm();
+    UT_LOG_DEBUG("Return status: %d", status);
+    UT_ASSERT_EQUAL_FATAL(status, tvERROR_NONE);
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief This test aims to verify the functionality of setting and getting the Gamma Table in the TV settings.
+*
+* In this test, the SetGammaTable() and GetGammaTable() functions are tested to
+* ensure they are working as expected. The test first initializes the TV, sets the Gamma
+* Table, retrieves it, and then compares the set and retrieved values to ensure they match.
+* If any of these steps fail, the test is considered failed.
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 035@n
+*
+* **Test Procedure:**
+* Refer to Test specification documentation [tv-settings_L2_Low_Level_Test_Spec.md](../docs/pages/tv-settings_L2_Low_Level_Test_Spec.md)
+*/
+
+void test_l2_tvSettings_SetAndGetGammaTable(void)
+{
+    gTestID = 35;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    tvError_t ret = tvERROR_NONE;
+    uint16_t size = GAMMA_TABLE_SIZE;
+    uint16_t *pData_R = malloc(size * sizeof(uint16_t));
+    uint16_t *pData_G = malloc(size * sizeof(uint16_t));
+    uint16_t *pData_B = malloc(size * sizeof(uint16_t));
+    uint16_t *getData_R = malloc(size * sizeof(uint16_t));
+    uint16_t *getData_G = malloc(size * sizeof(uint16_t));
+    uint16_t *getData_B = malloc(size * sizeof(uint16_t));
+    uint32_t rValue[GAMMA_TABLE_SIZE] = {
```

Comment:
Hi:
Where do these values come from (lines 2499-2549)?  Are they original to you or from an external specification?  There is a code match which is causing complications in the tool, but it may not be real.
Thank you,
Martin Hughes, CMF Compliance Team

Reply:
The values are random and original to me, added to test the scenario, and they are not from any external specification.

Reply:
HI @mhughesacn, can you please resolve this if you are Ok with @Anbukannadhasan reply?
Thanks.